src/hex_view.rs

```diff
@@ -145,17 +145,19 @@ impl HexView {
         if self.pos_locked {
             return;
         }
-        self.cur_pos = val.clamp(0, self.file.data.len() - self.bytes_per_row);
+        let last_line_start_address =
+            (self.file.data.len() / self.bytes_per_row) * self.bytes_per_row;
```

Comment:
The last line address is `data_len / bytes_per_row` floored and then multiplied by `bytes_per_row` to get the number of bytes until the first byte in the last line.

The calculation could also be `data_len - (data_len % bytes_per_row)`, which could be more performant, but the performance difference will be negligible here.